src/metrics/include/metrics/v2/event.hpp

```diff
@@ -137,13 +155,28 @@ class StreamInterfaceImpl {
     }
 
 private:
+    bool finish_call() {
+        if (_in_flight) {
+            void* got_tag;
+            bool ok = false;
+            _cq.Next(&got_tag, &ok);
+
+            _in_flight = false;
+            return got_tag == _tag && ok;
+        }
+        return true;
+    }
+
     std::string _name;
     ActorId _actorId;
+    bool _in_flight;
     CollectorStubInterface _stub;
     grpc::WriteOptions _options;
     poplar::PoplarResponse _response;
     grpc::ClientContext _context;
-    std::unique_ptr<grpc::ClientWriterInterface<poplar::EventMetrics>> _stream;
+    grpc::CompletionQueue _cq;
+    void* _tag;
```

Comment:
while not incorrect this name isn't very..obvious. Instead maybe `_grpcInFlight`? I think it can also be `const` but actually just realizing the rest of this isn't const so prolly best to not shake it up.